### ISSUE TITLE
WP_CLI::runcommand() reusing the process does not attach media to a post

### ISSUE THREAD

**Issue by slavicd** [wp-cli/wp-cli#6025]
## Bug Report

This results in an attachment being created, but not attached to any particular post.

```PHP
WP_CLI::runcommand(
            sprintf('media import "%s" --post_id="%d" --featured_image --porcelain', $img_path, $post_id),
            ['return' => true, 'launch' => false]
        );
```

This results in an attachment being created, and attached to the post (**expected behaviour**).

```PHP
WP_CLI::runcommand(
            sprintf('media import "%s" --post_id="%d" --featured_image --porcelain', $img_path, $post_id),
            ['return' => true, 'launch' => true]
        );
```

**Let us know what environment you are running this on**

```
wp cli info
OS:	Linux 5.15.0-1061-gke #67-Ubuntu SMP Wed Jun 12 18:17:06 UTC 2024 x86_64
Shell:	/bin/bash
PHP binary:	/usr/bin/php8.2-zend
PHP version:	8.2.24
php.ini used:	/etc/php/8.2-zend/cli/php.ini
MySQL binary:	/usr/bin/mysql
MySQL version:	mysql  Ver 8.0.39-0ubuntu0.22.04.1 for Linux on x86_64 ((Ubuntu))
SQL modes:	STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION
WP-CLI root dir:	phar://wp-cli.phar/vendor/wp-cli/wp-cli
WP-CLI vendor dir:	phar://wp-cli.phar/vendor
WP_CLI phar path:	/nas/content/live/rcmathewsdev
WP-CLI packages dir:	
WP-CLI cache dir:	/home/wpe-user/.wp-cli/cache
WP-CLI global config:	/home/wpe-user/.wp-cli/config.yml
WP-CLI project config:	
WP-CLI version:	2.11.0

```

**Comment by karthick-murugan** [comment]
Hi @slavicd.,

As mentioned by @swissspidy, here is the video where I attempted to reproduce the previously mentioned error. However, I am now unable to replicate the issue—whether launching with true or false, the media uploads successfully and links to the post both times.

Please review the video and let me know if I am missing anything, or share the exact steps to reproduce the error. Looking forward to your insights.

https://github.com/user-attachments/assets/7de0dda0-7c1d-4ead-b251-59553cafc215

**Comment by swissspidy** [comment]
Thanks for sharing!

The text from the video for anyone trying to reproduce:

`WP_CLI::runcommand('media import "/var/www/html/wp-content/sardinian-bream.jpg" --post_id="1" --featured_image --porcelain', ['launch'=>false]);`

The key bit is the warning that's being printed.

It happens because `--post_id="1"` is parsed as a literal string `"1"` with quotes, hence the warning.

Soo.... have you tried removing the quotes?

----

This parsing happens in `Configurator::extract_assoc()`. Not sure if this could or should be modified to handle this `runcommand()` edge case.

**Comment by slavicd** [comment]
> Soo.... have you tried removing the quotes?

Yes, the video has 3 cases: one with quotes in a separate process, one with quotes in the same process (failing) and one without quotes.

This bug demonstrates different behaviour based on whether the command runs in the same process or not, so at least from this perspecive, it needs to be addressed. Also, as far as the warning goes (which by the way is not there when you use return=>true), my expectation would be that `:runcommand()` should behave the same as the actual command inside would.
